### PR TITLE
Problem: missing cronos NFT minting (fix #458)

### LINKED ISSUE #458: Problem: missing cronos NFT  minting
add cronos NFT mint, issue apis
erc20, erc721 , erc1155

this will be used for NFT generation tool

### PR DESCRIPTION
**Pull request by leejw51crypto**
Solution: add deploying, minting example

### CODE REVIEW

**Review by tomtau — CHANGES_REQUESTED** [review]
a few clippy errors: https://github.com/crypto-com/defi-wallet-core-rs/actions/runs/3617711970/jobs/6096806593#step:6:162

File: bindings/cpp/src/ethereum.rs [hunk 1]
```diff
@@ -63,6 +65,18 @@ mod ffi {
...
 
+        // extract toplevel json with keyname
+        fn extract_json(src: String, keyname: String) -> Result<String>;
+        // 0x... -> bytes
+        fn extract_bytes(src: String, keyname: String) -> Result<Vec<u8>>;
+
+        fn encode_deploy_contract(
+            rpcserver: String,
+            abi: String,
+            bytecode: Vec<u8>,
```
Comment (damoncro) [inline bindings/cpp/src/ethereum.rs:76]: These twos are actually from the abi file, how about make it just as simple as possible - read abi file by name, and do them in Rust, to replace std::ifstream to read file then extract_json, extract_bytes etc C++.

Comment (damoncro) [inline bindings/cpp/src/ethereum.rs:76]: These twos are actually from the abi file, how about make it just as simple as possible - read abi file by name, and do them in Rust, to replace std::ifstream to read file then extract_json, extract_bytes etc C++.

**Review by damoncro — DISMISSED** [review]

File: bindings/cpp/src/ethereum.rs (hunk 1 above)
Comment (leejw51crypto) [reply bindings/cpp/src/ethereum.rs]: ok

File: example/cpp-example/cronos.cc [hunk 2]
```diff
@@ -638,6 +637,54 @@ void test_dynamic_api_send() {
...
+  eth_tx_info.amount_unit = EthAmount::EthDecimal;
+  eth_tx_info.data = data;
+  eth_tx_info.gas_limit = "1955020";
+  eth_tx_info.gas_price = "100000";
+  eth_tx_info.gas_price_unit = EthAmount::WeiDecimal;
+
+  Vec<uint8_t> signedtx =
+      build_eth_signed_tx(eth_tx_info, chainid, true, *privatekey);
+  CronosTransactionReceiptRaw receipt =
+      broadcast_eth_signed_raw_tx(signedtx, mycronosrpc, 1000);
```
Comment (tomtau) [inline example/cpp-example/cronos.cc:679]: does this need to be specified manually? why not just `w->send(...)` ?

Comment (leejw51crypto) [reply example/cpp-example/cronos.cc:679]: send will work, too
this is for display for wallet-connect too.
in gamefi, usually, wallet-connect will be used

**Review by tomtau — APPROVED** [review]

**Pull request merged**